lib/pronto/message.rb

```diff
@@ -4,17 +4,19 @@ class Message
 
     LEVELS = [:info, :warning, :error, :fatal]
 
-    def initialize(path, line, level, msg, commit_sha = nil)
-      unless LEVELS.include?(level)
+    def initialize(options)
+      unless LEVELS.include?(options[:level])
         raise ::ArgumentError, "level should be set to one of #{LEVELS}"
       end
 
-      @path = path
-      @line = line
-      @level = level
-      @msg = msg
-      @commit_sha = commit_sha
-      @commit_sha ||= line.commit_sha if line
+      @path = options[:path]
+      @line = options.fetch(:line, nil)
+      @level = options[:level]
+      @msg = options[:msg]
+      @runner = options.fetch(:runner, nil)
```

Comment:
Yup, but then the number of arguments will trigger a rubocop warning, is that okay?